public/app/features/plugins/routes.tsx

```diff
@@ -4,26 +4,28 @@ import { getRootSectionForNode } from 'app/core/selectors/navModel';
 import AppRootPage from 'app/features/plugins/components/AppRootPage';
 import { getState } from 'app/store/store';
 
+const isPluginNavModelItem = (model: NavModelItem): model is PluginNavModelItem => 'pluginId' in model && 'id' in model;
+const isStandalonePluginPage = (id: string) => id.startsWith('standalone-plugin-page-/');
+
+function getPathForNavItem(navItem: PluginNavModelItem) {
+  const pluginNavSection = getRootSectionForNode(navItem);
+  const appPluginUrl = `/a/${navItem.pluginId}`;
+  const path = isStandalonePluginPage(navItem.id) ? navItem.url || appPluginUrl : appPluginUrl; // Only standalone pages can use core URLs, otherwise we fall back to "/a/:pluginId"
+  const isSensitive = isStandalonePluginPage(navItem.id) && !navItem.url?.startsWith('/a/'); // Have case-sensitive URLs only for standalone pages that have custom URLs
+
+  return {
+    path: `${path}/*`,
+    sensitive: isSensitive,
+    component: () => <AppRootPage pluginId={navItem.pluginId} pluginNavSection={pluginNavSection} />,
+  };
```

Comment:
One thing I'm curious about...there are some `RouteDescriptor` properties that we don't currently return. Notably, `roles` and `chromeless`. I'd like to understand why.